public/app/core/components/OptionsUI/DashboardPickerByID.tsx

```diff
@@ -70,10 +71,12 @@ async function getDashboards(
   label: string,
   excludedDashboards?: string[]
 ): Promise<Array<SelectableValue<DashboardPickerItem>>> {
-  const result = await backendSrv.search({ type: 'dash-db', query, limit: 100 });
+  // FIXME: stop using id from search and use UID instead
```

Comment:
in https://github.com/grafana/grafana/pull/54125 we can remove this component entirely 🎉